python/fledge/plugins/filter/rename/rename.py

```diff
@@ -166,14 +166,31 @@ def find_and_replace(operation, find, replace_with, ignore_case, reading):
         dict:          A processed dictionary
     """
     revised_reading_dict = {}
-    _flag = re.IGNORECASE if ignore_case == 'false' else False
-
-    def get_all_values(nested_dictionary):
-        for key, value in nested_dictionary.items():
-            if type(value) is dict:
-                get_all_values(value)
-            _datapoint = re.sub(search_pattern, replace_with, str(key), flags=_flag)
-            revised_reading_dict[_datapoint] = value
+    _ignore_case_flag = re.IGNORECASE if ignore_case == 'false' else False
```

Comment:
Plugin handle for boolean types are always in string format. i.e `"ignoreCase": {"value": "false"}`
If we want to change the condition here, then we need to first it convert to bool type and then set accordingly.

Reply:
Actually my comment is for assignment to ignore case flag variable.

Anyways, todo can be removed later; and if functional okay then we can merge this PR.

Reply:
It would be better ` _ignore_case_flag = re.IGNORECASE if ignore_case == 'false' else 0`

As python-regex flag value should be an INTEGER as it derived from `enum.IntFlag`. And for every python regex function its (flag) default value is 0.

FYI,
```
re.IGNORECASE
<RegexFlag.IGNORECASE: 2>
```
And for functionality wise I have added my notes, I see issues with the change.